pygls/protocol.py

```diff
@@ -78,12 +79,33 @@ def decorator(self, *args, **kwargs):
 
 def dict_to_object(**d):
     """Create nested objects (namedtuple) from dict."""
+    type_name = d.pop('type_name', 'Object')
     return json.loads(
         json.dumps(d),
-        object_hook=lambda p: namedtuple('Object', p.keys(), rename=True)(*p.values())
+        object_hook=lambda p: namedtuple(type_name, p.keys(), rename=True)(*p.values())
     )
 
 
+def default_serializer(o):
+    """JSON serializer for complex objects."""
+    if isinstance(o, enum.Enum):
+        return o.value
+    return o.__dict__
+
+
+def deserialize_command(params):
+    """Function used to deserialize command arguments to a specific class
+    or a namedtuple."""
+    cmd_name = params['command']
+    # TODO: Register/Look up custom command arguments' types
+
+    type_name = re.sub('[^a-zA-Z]+', '', cmd_name) + 'Params'
```

Comment:
I understand the general idea, but sure that I understand what needs to be done here. Can you add a comment and provide an example of input and expected output?

Reply:
Updated